File: leetcode/14-longest-common-prefix.py
# https://leetcode.com/problems/longest-common-prefix/ (top-interview-questions)
from typing import List

def longestCommonPrefix(strs: List[str]) -> str:
    min_str = strs[0]
    for string in strs:
        if len(string) < len(min_str): min_str = string

    prefix = min_str

    for string in strs:
        for i in range(len(min_str)):
            if i < len(prefix) and prefix[i] != string[i]:
                prefix = prefix[:i]
                break

    return prefix

File: leetcode/14-longest-common-prefix.js
// https://leetcode.com/problems/longest-common-prefix/ (top-interview-questions)

/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let minStr = strs[0];
    
    for (let i = 1; i < strs.length; i++) {
      if (minStr.length > strs[i].length) {
        minStr = strs[i];
      }
    }
  
    let commPrefix = "";
  
    for (let i = 0; i < minStr.length; i++) {
      let j = 0;
      while (j < strs.length && minStr[i] === strs[j][i]) j += 1;
      if (j === strs.length) commPrefix += minStr[i];
      else break;
    }
  
    return commPrefix;
  
  };